### ISSUE TITLE
[Bug]: NuGet Package Manager does not update Directory.Packages.props

### ISSUE THREAD

**Issue by sbenzenko** [NuGet/Home#11826]
### NuGet Product Used

Visual Studio Package Management UI
Visual Studio Package Manager Console

### Product Version

dotnet 6.0.300, Visual Studio 17.2.1

### Impact

It's more difficult to complete my work

### Repro Steps & Context

When using Central Package Management with `Directory.Packages.props` file on solution level, trying to update package from _Visual Studio Package Management UI_ or via _Package Management Console_ results in package version being updated in project's `.csproj` file rather than in `Directory.Packages.props` file. It adds `Version="..."` attribute to `<PackageReference />` in `.csproj`. That in turn causes the following compilation error:
`NU1008	Projects that use central package version management should not define the version on the PackageReference items but on the PackageVersion items`

Had to manually remove `Version="..."` from `.csproj` and update it in `Directory.Packages.props` to make it work.
![image](https://user-images.githubusercontent.com/7808628/169713338-e2aa6626-c6c7-45e5-80e7-c03c947d68fc.png)

**Comment by jeffkl** [comment]
@sbenzenko Thank you for your report, we are actively working on Visual Studio support as well as `dotnet` CLI support for CPM.

At a top level, the Visual Studio work is tracked here: https://github.com/NuGet/Home/issues/9015

**Comment by reduckted** [comment]
@jeffkl Does that mean we can't use Central Package Management _and_ install new packages in Visual Studio? 

I've switched to CPM and Visual Studio can _restore_ the packages just fine, but when I try to install a package, I get the error mentioned in this issue _and_ the package installation is rolled back, so I can't install any packages at all.

```
NU1008: Projects that use central package version management should not define the version on the PackageReference items but on the PackageVersion items: MyPackage.
Package restore failed. Rolling back package changes for 'MyProject'.
```

That seems to go against what [the documentation](https://docs.microsoft.com/en-us/nuget/consume-packages/central-package-management) says about this feature:

> The feature is available across all NuGet integrated tooling.
> * [Visual Studio 2022 17.2 and later](https://visualstudio.microsoft.com/downloads/)

**Comment by jeffkl** [comment]
> @jeffkl Does that mean we can't use Central Package Management and install new packages in Visual Studio?

@reduckted That's correct.  We are working on the tooling side of things like Visual Studio and `dotnet` so that they work with CPM.  Some of the work is already done and will be available in upcoming releases.  We're tracking the work here: https://github.com/NuGet/Home/issues/11752

Sorry that wasn't clear from the documentation.

**Comment by awattar** [comment]
@jeffkl any ETA for this basic functionality?

**Comment by yorschor** [comment]
This is also a Problem when using .Net Framework projects with the legacy project format and Visual Studio. From reading the [CPM](https://learn.microsoft.com/en-us/nuget/consume-packages/central-package-management) docu this should work. 

> "Central package management applies to all <PackageReference>-based MSBuild projects (including [legacy CSPROJ](https://github.com/dotnet/project-system/blob/main/docs/feature-comparison.md)) as long as compatible tooling is used."

Restoring works. Upgrading via the Rider NugGet manager works. Upgrading with the Visual Studio tools does not :( 
![image](https://github.com/user-attachments/assets/3adc71aa-a14a-4ccb-ab61-41c5d33625d2)

---
Visual Studio 17.10.5
.NET Framework 4.6.2 & 4.7.2

**Comment by kcamp** [comment]
@nkolev92 Is there any update on this issue or guess on timeline for implementation & resolution?  The inability to effectively manage CPM in VS for legacy [web] projects utilizing `<PackageReference>` as described in https://github.com/NuGet/Home/issues/11826#issuecomment-2273587822 is a big gap in tooling that is preventing broader adoption of CPM by our teams.  Thanks!

**Comment by nkolev92** [comment]
SDK Style projects are supported.  https://github.com/NuGet/Home/issues/9015

Legacy is more challenging and we have a tracking issue in https://github.com/NuGet/Home/issues/9521.

I'll consolidate those issues. 

When there's updates, we'll provide https://github.com/NuGet/Home/issues/9521.

In whichever scenarios possible, please consider migrating your classic csproj PackageReference to SDK style. There's benefits there that go beyond just CPM, especially if you already have .NET SDK based projects in your solution.
